File: tests/fixtures/supabase/supabase-storage-error-not-surfaced-broken/01-avatar-upload.ts
import { supabase } from '../client.js';

export async function saveAvatar(userId: string, file: File, avatarUrl: string) {
  const path = `${userId}/avatar.png`;
  const { error: uploadError } = await supabase.storage.from('avatars').upload(path, file);
  if (!uploadError) {
    const { data } = supabase.storage.from('avatars').getPublicUrl(path);
    avatarUrl = data.publicUrl;
  }
  await supabase.from('profiles').update({ avatar_url: avatarUrl }).eq('user_id', userId);
}

File: tests/fixtures/supabase/supabase-storage-error-not-surfaced-broken/01_avatar_upload.py
from lib.client import supabase


def save_avatar(user_id: str, file, avatar_url: str):
    path = f"{user_id}/avatar.png"
    try:
        supabase.storage.from_("avatars").upload(path=path, file=file)
    except Exception:
        # Swallowed — execution falls through to the profile update below
        # even though the file never actually uploaded.
        pass
    supabase.table("profiles").update({"avatar_url": avatar_url}).eq("user_id", user_id).execute()
